### PR TITLE
Adds testable examples to be automatically pulled in redis.io docs

### PR DESCRIPTION
**Pull request by elena-kolevska**
These examples appear on the [Redis documentation](https://redis.io/commands/set/) site as part of the tabbed examples interface.

### CODE REVIEW

**Comment by monkey92t** [comment]
👍🏻

File: .github/workflows/doctests.yaml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        image: redis/redis-stack-server:latest
+        options: >-
+          --health-cmd "redis-cli ping" --health-interval 10s --health-timeout 5s --health-retries 5
+        ports:
+          - 6379:6379
+
+    strategy:
+      fail-fast: false
+      matrix:
+        go-version: [ 1.18.x, 1.19.x, 1.20.x ]
```
Comment (chayim) [inline .github/workflows/doctests.yaml:28]: IMHO for the purposes of the doctests, you can rely on the oldest. 1.18 -> 1.20 is fine for these. I don't think we need that degree of coverage, as it's duplication.

File: .github/workflows/doctests.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
+name: Documentation Tests
+
+on:
+  push:
+    branches: [master, v9]
+  pull_request:
+    branches: [master, v9]
```
Comment (chayim) [inline .github/workflows/doctests.yaml:7]: For now, let's master and your branch only.

File: .github/workflows/doctests.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      - name: Set up ${{ matrix.go-version }}
+        uses: actions/setup-go@v4
+        with:
+          go-version: ${{ matrix.go-version }}
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Test doc examples
+        working-directory: ./doctests
```
Comment (chayim) [inline .github/workflows/doctests.yaml:40]: Glad we're using this tree!

File: doctests/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
+# Command examples for redis.io
+
+These examples appear on the [Redis documentation](https://redis.io) site as part of the tabbed examples interface.
+
+## How to add examples
+
+- Create a Go test file with a meaningful name in the current folder. 
+- Create a single method prefixed with `Example` and write your test in it.
+- Find out what's the example id for the example you're creating and add it as the first line of the file `// EXAMPLE: set_and_get`. See more info below under the [Example](#example) section.
```
Comment (chayim) [inline doctests/README.md:9]: ```Determine the id for the example you're creating and add it as the first line of ...```

```See below in Example](#example) section.

File: doctests/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+- We're using the [Testable Examples](https://go.dev/blog/examples) feature of Go to test the desired output has been written to stdout.
+- To run the tests start a Redis server locally on port 6379 and run `go test`. (This will be improved and aligned with the way we run the main test suite).
```
Comment (chayim) [inline doctests/README.md:11]: Start a Redis server locally, on port 6379 and run `go test`.

Comment (chayim) [reply doctests/README.md:11]: I might add - this is a duplication, since how to add and how to test contain similar content. Reduce?

**Comment by elena-kolevska** [comment]
Thanks for the review @chayim. It should be good now.

**Review by chayim — APPROVED** [review]

**Pull request merged**